mlflow/server/js/src/components/ExperimentPage.js

```diff
@@ -23,42 +25,61 @@ class ExperimentPage extends Component {
     dispatchSearchRuns: PropTypes.func.isRequired,
   };
 
-  state = {
-    paramKeyFilter: new KeyFilter(),
-    metricKeyFilter: new KeyFilter(),
+  static defaultState = {
+    paramKeyFilterString: "",
+    metricKeyFilterString: "",
     getExperimentRequestId: getUUID(),
     searchRunsRequestId: getUUID(),
     searchInput: '',
     lastExperimentId: undefined,
     lifecycleFilter: LIFECYCLE_FILTER.ACTIVE,
   };
 
+  static getStore(experimentId) {
+    return LocalStorageUtils.getStore("ExperimentPage", experimentId);
+  }
+
+  store = ExperimentPage.getStore(this.props.experimentId);
+  state = LocalStorageUtils.loadComponentState(this.store, ExperimentPage.defaultState);
+
+  /** Wrapper over setState that caches certain fields in local storage. */
+  setStateWrapper(newState) {
```

Comment:
Could we do this on `componentDidUpdate` instead of introducing a wrapper? It seems un-React-like to have to put intermediate methods between code and updating the state -- it would be easily to accidentally forget to invoke this method if we added a new state change method, for example.